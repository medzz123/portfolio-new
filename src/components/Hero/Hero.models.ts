/**
 * Hero component for most pages
 *
 * If children is passed then only expanded prop will work
 */
export type HeroProps = (HeroWithChildren | HeroWithCustomProps) & {
  /**
   * Expands the container to occupy the full height of the viewport
   * @defaults false
   */
  expand?: boolean;
};

export interface HeroWithChildren {
  children?: React.ReactNode;
  title?: never;
  description?: never;
  date?: never;
  image?: never;
  tags?: never;
}

export interface HeroWithCustomProps {
  children?: never;
  title?: string;
  description?: string;
  date?: string;
  image?: string;
  tags?: string[];
}
