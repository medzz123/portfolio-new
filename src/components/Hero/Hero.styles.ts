import { styled } from '@theme/config';

export const HeroContainer = styled('div', {
  background:
    'rgba(0, 0, 0, 0) linear-gradient($secondaryBackground 80%, transparent 10%, transparent 10%) repeat scroll 0% 0%',
  margin: '$24',
  center: 'column',
  padding: '$20',

  variants: {
    expand: {
      true: {
        backgroundColor: '$secondaryBackground',
        height: '$full',
      },
    },
  },
});

export const CustomContainer = styled('div', {
  margin: 'auto',
});

export const Img = styled('img', {
  maxWidth: '$6xl',
  width: '$full',
  height: 'auto',
});
