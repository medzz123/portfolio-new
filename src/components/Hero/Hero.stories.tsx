import { Meta, Story } from '@storybook/react/types-6-0';
import React from 'react';

import Hero from './Hero';

export default {
  title: 'Components/Hero',
  component: Hero,
} as Meta;

const Template: Story = (storyArguments) => <Hero {...storyArguments} />;

export const Default = Template.bind({});
