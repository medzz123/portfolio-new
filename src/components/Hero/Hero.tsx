import { Box } from '@components/Box';
import Layout from '@components/Layout';
import { H1, P } from '@components/Text';
import React, { FunctionComponent } from 'react';

import { HeroProps } from './Hero.models';
import { CustomContainer, HeroContainer } from './Hero.styles';

const Hero: FunctionComponent<HeroProps> = (props) => {
  const { expand, children, title, image, description, date } = props;

  return (
    <HeroContainer expand={expand} data-testid="hero">
      <Layout>
        {children ?? (
          <CustomContainer>
            <H1 align="center">{title}</H1>
            <P align="center">{date}</P>
            <P align="center">{description}</P>
            <Box css={{ mb: '$80' }} />
            <img src={image} width="100%" alt="Showcase" css={{ borderRadius: '$8' }} />
          </CustomContainer>
        )}
      </Layout>
    </HeroContainer>
  );
};

export default Hero;
