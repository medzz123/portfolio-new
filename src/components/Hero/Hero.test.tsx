import { render } from '@test/library';
import React from 'react';

import Hero from './Hero';

test('Hero Test', () => {
  const { getByTestId, container } = render(<Hero>Hello</Hero>);

  const heroNode = getByTestId('hero');

  expect(heroNode.textContent).toBe('Hello');

  expect(container).toMatchSnapshot();
});
