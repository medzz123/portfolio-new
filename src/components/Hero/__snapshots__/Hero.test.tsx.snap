// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hero Test 1`] = `
<div>
  <div
    class="sxodqir"
    data-testid="hero"
  >
    <div
      class="sxsw5wg"
      data-testid="layout"
    >
      Hello
    </div>
  </div>
</div>
`;
